file_windows.go

```diff
@@ -0,0 +1,16 @@
+package txfile
+
+// computePlatformMmapSize computes the maximum amount of bytes to be mmaped,
+// depending on the actual file size and the configured maximum file size.
+// On Windows, the size returned MUST NOT exceed the actual file size.
+func computePlatformMmapSize(fileSize, maxSize, pageSize uint) (uint, error) {
+	if maxSize == 0 {
+		return fileSize, nil
+	}
```

Comment:
What is the purpose of this check? This look like some sort of capping or fallback? And why we only target the windows platform?

Reply:
Windows and Unix mmap work somewhat differently. Windows requires us to never exceed the actual file size. If the file is pre-allocated, computeMmapSize is doing the right size. But if not or `maxSize == 0`, we must cap on the actual file size. => without prealloc we need to unmmap and mmap much more often.